2017/proposals/patni-harshit.md

```diff
@@ -0,0 +1,176 @@
+# Title
+2D color maps
+
+## Abstract
+
+All of the color mapping in Matplotlib is currently derived from
+`ScalerMappable` which as the name suggests maps scalers from `R^1 ->
+R^4` RGBA color space.  It is common to want to map a vector to
+colors, for example to control the alpha based on a second value in a
+scatter plot or to show the orientation of a field.
+
+| **Intensity** | **Involves**  | **Mentors** |
+| ------------- | --------------|------------ |
+| Intermediate  | Python, | [@tacaswell][@story645] |
+
+## Technical Details
+
+- All the work will be done in Python.
+- It will be in accordance with current implementation of 1D colormaps.
+- Extending the existing 1D normalization for 2D data : This can be done by
+a family of normalizers which go from data -> unit-disk or square in the
+complex plane
+- Creating color maps that go from the unit disk or square -> RGBA
+- Exposing these classes to user as either new API or extending the existing
+`ScalerMappable` API
+- Implementing a 2D color bar
+- Developing 2D color maps : They can be created by doing bi-linear
+interpolation between four colours. Also major focus would be on developing
+perpetual colormaps so that color blind people have no difficulty in
+distinguishing colors.
+
+## Schedule of Deliverables
+
+### May 1th - May 28th, **Community Bonding Period**
+
+- Before the official time period begins I will do some tasks listed under
+[MEP21](http://matplotlib.org/devel/MEP/MEP21.html): color and cm refactor.
+This will greatly help in understanding the current implementation of
+normalization and color mapping tools in Matplotlib
+- Set up a blog
+- Along with this I will continue to solve issues on github
+
+### May 29th - June 3rd
+
+- Discuss implementation details with mentors
+- Start working on normalizers
+
+### June 5th - June 9th
+
+- Finish up normalizers
+- Write tests
+- Starting working on color map that maps unit circle or square to rgba
+
+### June 12th - June 16th
+
+- Finish up color maps
+- Debug and test
+- Write tests
+
+### June 19th - June 23th, **End of Phase 1**
+
+- Complete any unfinished work in Phase 1
+- Write documentation for code written so far
+- Write blog for Phase 1
+
+### June 26 - June 30th, **Begin of Phase 2**
+
+- Decide on how API will be exposed to users as new API or as extension of
+ScalerMappable
+- Make API for exposing normalizers and color maps to user
+
+### July 3rd - July 7th
+
+- Test new API
+- Document the API so that it is exposed to users
+
+### July 10th - July 14th
+
```

Comment:
This concerns me a bit because the colorbar is a ton of work and will have a good amount of community input, so I strongly suggest allocating more time for it.